crates/sequencer/src/state.rs

```diff
@@ -14,28 +17,58 @@ use starknet_api::{
     hash::StarkFelt,
 };
 
+use serde::{Deserialize, Serialize};
+
 use crate::commit::Committer;
+use crate::serde::{SerializableState, SerializationError};
 
 /// Generic state structure for the sequencer.
 /// The use of `FxHashMap` allows for a better performance.
 /// This hash map is used by rustc. It uses a non cryptographic hash function
 /// which is faster than the default hash function. Think about changing
 /// if the test sequencer is used for tests outside of ef-tests.
 /// See [rustc-hash](https://crates.io/crates/rustc-hash) for more information.
-#[derive(Default)]
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
 pub struct State {
-    classes: FxHashMap<ClassHash, ContractClass>,
-    compiled_class_hashes: FxHashMap<ClassHash, CompiledClassHash>,
-    contracts: FxHashMap<ContractAddress, ClassHash>,
-    storage: FxHashMap<ContractStorageKey, StarkFelt>,
-    nonces: FxHashMap<ContractAddress, Nonce>,
+    pub classes: FxHashMap<ClassHash, ContractClass>,
+    pub compiled_class_hashes: FxHashMap<ClassHash, CompiledClassHash>,
+    pub contracts: FxHashMap<ContractAddress, ClassHash>,
+    pub storage: FxHashMap<ContractStorageKey, StarkFelt>,
+    pub nonces: FxHashMap<ContractAddress, Nonce>,
```

Comment:
mmmm do we need to expose all this? is pub(crate) not enough?

Reply:
resolved via [this](https://github.com/kkrt-labs/ef-tests/pull/555/commits/47a3e76278392d01c00a97ffed65d9b34d1f84d0) commit.